embedded/__main__.py

```diff
@@ -60,30 +81,162 @@ def handle_gps_polling():
 def handle_lighting():
     """Handles the lighting state management."""
 
-    status_led = RGBLED(13, 19, 26)
     steps = 100
     current_step = 0
 
     while not QUIT_EVENT.is_set():
         if GPS_STATUS in GPSStatus.locked_states():
-            set_rgb_colour(status_led, Colour.GREEN)
+            print(GPS_MODULE.get_location())
+            #Setting bike according to location status
+            if bike.locked == true:
+                set_rgb_colour(status_led, Colour.RED)
+            else:
+                set_rgb_colour(status_led, Color.GREEN)
             sleep(1)
         else:
+            #Cycles if the GPS does not have a lock
             current_step = (current_step + 1) % steps
             cycle_rgb_led(status_led, current_step, steps)
             sleep(1 / steps)
+    close()
 
+def close():
+    QUIT_EVENT.set()
+    for thread in threads:
+        thread.join()
     status_led.off()
     status_led.close()
+    GPS_MODULE.close()
+
+   # gpsModule.close
+   # status.close
+
+
+async def create_ticket(session, bike) -> bytes:
+    """
+        Gets the challenge from the server.
+
+        :return: The challenge bytes
+        :raises AuthError:
+        """
+    async with session.post(URL + "/connect", data=bike.public_key.encode(RawEncoder)) as resp:
+        if resp.status != 200 and resp.reason == "Identity not recognized.":
+            raise AuthError("public key not on server")
+        return await resp.read()
+
+
+def bike_handler(session, bike, signature):
```

Comment:
Get rid of this line and unindent. It's returning null because the bike_handler function declares an inner function and returns nothing.